plugins/eos/gs-plugin-eos.c

```diff
@@ -1374,3 +1374,56 @@ gs_plugin_add_updates (GsPlugin *plugin,
 	 * this will still work correctly */
 	return add_updates (plugin, list, cancellable, error);
 }
+
+gboolean
+gs_plugin_add_popular (GsPlugin *plugin,
+		       GsAppList *list,
+		       GCancellable *cancellable,
+		       GError **error)
+{
+	g_autoptr(AsProfileTask) ptask = NULL;
+	GsAppList *new_list = NULL;
+	const gchar *popular_apps[] = {"com.google.Chrome.desktop",
+				       "com.spotify.Client.desktop",
+				       "com.transmissionbt.Transmission.desktop",
+				       "com.valvesoftware.Steam.desktop",
+				       "libreoffice-calc.desktop",
+				       "libreoffice-impress.desktop",
+				       "libreoffice-writer.desktop",
+				       "net.minetest.Minetest.desktop",
+				       "net.sourceforge.Audacity.desktop",
+				       "org.debian.alioth.tux4kids.Tuxmath.desktop",
+				       "org.gnome.GIMP.desktop",
+				       "org.inkscape.Inkscape.desktop",
+				       "org.mozilla.Firefox.desktop",
+				       "org.pitivi.Pitivi.desktop",
+				       "org.tuxpaint.Tuxpaint.desktop",
+				       "org.videolan.VLC.desktop",
+				       "simple-scan.desktop",
+				       NULL};
+
+	ptask = as_profile_start_literal (gs_plugin_get_profile (plugin),
+					  "eos::add-popular");
+
+	new_list = gs_app_list_new ();
+
+	/* add the hardcoded list of popular apps */
+	for (guint i = 0; popular_apps[i] != NULL; ++i) {
+		g_autoptr(GsApp) app = gs_app_new (popular_apps[i]);
+		gs_app_add_quirk (app, AS_APP_QUIRK_MATCH_ANY_PREFIX);
```

Comment:
What is the purpose of this quirk?  Would it be obvious to most developers, or should we add a comment?

Reply:
App IDs are composed of semantically different parts: scope (system/user), bundle type (package/flatpak/snap), etc.
Since we just set up the simple ID in the app, this quirk makes sure that the app will match e.g. all org.gimp.GNOME.desktop apps. In the context of GNOME Software and AppStream-glib it's not very complicated, so I don't think we need a comment for this (otherwise we'll fill the code with such comments).